nautobot/ipam/prefix_ui.py

```diff
@@ -0,0 +1,139 @@
+            "tenant": getattr(obj.tenant, "pk", None),
+        }
+        if obj.locations.exists():
+            params["locations"] = [loc.pk for loc in obj.locations.all()]
+
+        return f"{reverse(self.link_name)}?{urlencode(params)}"
+
+
+class AddIPAddressButton(Button):
+    """Custom button to add an IP address inside a Prefix detail view."""
+
+    def should_render(self, context: Context):
+        if not super().should_render(context):
+            return False
+        return context.get("active_tab") == "ip-addresses" and context.get("first_available_ip") is not None
+
+    def get_link(self, context: Context):
+        from urllib.parse import urlencode
+
+        obj = get_obj_from_context(context)
+        first_available_ip = context.get("first_available_ip")
+        if not first_available_ip:
+            return None
+
+        params = {
+            "address": first_available_ip,
+            "namespace": getattr(obj.namespace, "pk", None),
+            "tenant_group": getattr(getattr(obj.tenant, "tenant_group", None), "pk", None),
+            "tenant": getattr(obj.tenant, "pk", None),
+        }
+        return f"{reverse(self.link_name)}?{urlencode(params)}"
+
+
+class PrefixKeyValueOverrideValueTablePanel(KeyValueTablePanel):
+    """A table panel for displaying key-value pairs of prefix-related attributes, along with any override values defined on the prefix object."""
+
+    def render_locations_list(self, key, value, instance):
+        """Renders a <ul> HTML list of locations with hyperlinks, or a placeholder if none exist."""
+        if not value or not value.exists():
+            return helpers.placeholder(None)
+
+        base_url = reverse("dcim:location_list")
+        full_listing_link = f"{base_url}?prefixes={instance.pk}"
+
+        return helpers.render_m2m(
+            value.all(),
+            full_listing_link=full_listing_link,
+            verbose_name_plural=key,
+        )
+
+    def render_utilization(self, value):
+        """Renders a utilization graph, or a placeholder if none exist."""
+        if value is None:
+            return helpers.placeholder(None)
+        return render_to_string(
+            "utilities/templatetags/utilization_graph.html",
+            helpers.utilization_graph(value),
+        )
+
+
+class PrefixObjectFieldsPanel(ObjectFieldsPanel, PrefixKeyValueOverrideValueTablePanel):
+    """
+    ObjectFieldsPanel that renders its fields in a 3-column layout.
+    Inherits behavior from ObjectFieldsPanel but overrides rendering with PrefixKeyValueOverrideValueTablePanel.
+    """
+
+    def get_data(self, context):
+        data = super().get_data(context)
+        fields = self.fields
+        instance = get_obj_from_context(context, self.context_object_key)
+
+        if instance and "utilization" in fields:
+            try:
+                data["utilization"] = instance.get_utilization()
+            except Exception:
```

Comment:
Do we know what cases would result in this exception? Missing code coverage suggests we aren't hitting any of them in the tests.